src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2652,6 +2670,9 @@ protected virtual void MarkInstruction (Instruction instruction, MethodDefinitio
 				}
 
 			case OperandType.InlineType:
+				if (instruction.OpCode.Code == Code.Newarr) {
+					Annotations.MarkRelevantToVariantCasting (((TypeReference) instruction.Operand).Resolve ());
```

Comment:
`Resolve` may return null - in which case we will add null to the `HashSet` in `Annotations`. I guess it doesn't matter that much, but it might be cleaner to check for null in the `MarkRelevantToVariantCasting` before adding it to the hashset.